src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java

```diff
@@ -0,0 +1,160 @@
+
+            dtoCount++;
+        }
+        reportWriterService.writeSubTitle(formatPropertyValue(CUPurapKeyConstants.JAGGAER_XML_REPORT_SECTION_FOOTER));
+    }
+
+    private String formatPropertyValue(String propertyName, String... args) {
+        String propertyValue = configurationService.getPropertyValueAsString(propertyName);
+        return MessageFormat.format(propertyValue, args);
+    }
+
+    private List<JaggaerUploadSupplierVendorDetailDto> filterDetailDto(List<JaggaerUploadSupplierVendorDetailDto> dtos, boolean hasNotes) {
+        return dtos.stream()
+                .filter(dto -> dto.hasNotes() == hasNotes)
+                .collect(Collectors.toList());
+    }
+
+    private void buildDetailSections(List<JaggaerUploadSupplierXmlFileDetailsDto> xmlFileDtos) {
+        xmlFileDtos.stream().forEach(dto -> {
+            reportWriterService.writeNewLines(4);
+            reportWriterService.writeSubTitle(formatPropertyValue(CUPurapKeyConstants.JAGGAER_XML_REPORT_SECTION_HEADER, dto.getXmlFileName()));
+            buildDetailLines(filterDetailDto(dto.getActiveVendors(), true));
+            buildDetailLines(filterDetailDto(dto.getActiveVendors(), false));
+            buildDetailLines(filterDetailDto(dto.getInactiveVendors(), true));
+            buildDetailLines(filterDetailDto(dto.getInactiveVendors(), false));
+            reportWriterService.writeSubTitle(formatPropertyValue(CUPurapKeyConstants.JAGGAER_XML_REPORT_SECTION_FOOTER));
+        });
+    }
+
+    private void buildDetailLines(List<JaggaerUploadSupplierVendorDetailDto> details) {
+        details.stream().forEach(detail -> {
+            reportWriterService.writeFormattedMessageLine(formatPropertyValue(CUPurapKeyConstants.JAGGAER_XML_REPORT_DETAIL_LINE,
+                        detail.isActive() ? ACTIVE : INACTIVE, detail.getVendorName(), detail.getVendorNumber()));
+            
+            detail.getNotes().stream().forEach(note -> 
+                reportWriterService.writeFormattedMessageLine(
+                        formatPropertyValue(CUPurapKeyConstants.JAGGAER_XML_REPORT_DETAIL_NOTE_LINE, note)));
+        });
+    }
+    
+    private void emailReport() {
+        final String toAddress = findReportToAddress();
+        final String fromAddress = toAddress;
+        final String body = LoadFileUtils.safelyLoadFileString(reportWriterService.getReportFile().getAbsolutePath());
+        
+        if (StringUtils.isNotBlank(toAddress)) {        
+            BodyMailMessage message = new BodyMailMessage();
+            message.setFromAddress(fromAddress);
+            String subject = reportWriterService.getTitle();
+            message.setSubject(subject);
+            message.getToAddresses().add(toAddress);
+            message.setMessage(body);
+    
+            boolean htmlMessage = false;
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("sendEmail, from address: " + fromAddress + "  to address: " + toAddress);
```

Comment:
There are several log statements in this method that need to update the mentioned method name: `sendEmail` --> `emailReport`.

Also, if desired, the debug statements could be parameterized and the call to `isDebugEnabled()` could be removed.

Reply:
ok